docs/specs/xcmp.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: default
+title: XCMP Interface
+parent: Specification
+nav_order: 2
+---
+
+# XCMP Interface <!-- omit in toc -->
+
+_NOTE: work in progress_
+
+- [Introduction](#introduction)
+- [Scenarios](#scenarios)
+  - [Alice transfers 21 PolkaETH to Bob on another chain](#alice-transfers-21-polkaeth-to-bob-on-another-chain)
+  - [Alice attempts to exchange 42 DOT for 21 PolkaETH](#alice-attempts-to-exchange-42-dot-for-21-polkaeth)
+- [Asset Identification](#asset-identification)
+- [Outstanding Issues](#outstanding-issues)
+  - [Numeric precision](#numeric-precision)
+
+## Introduction
+
+Our parachain will support cross-parachain token transfers using [XCMP](https://github.com/paritytech/xcm-format/blob/master/README.md).
+
+Since our parachain endeavours to be trustless, only reserve-backed transfers will be supported. Participating chains will need to agree on a parent chain to hold sovereign (reserve) accounts.
+
+## Scenarios
+
+### Alice transfers 21 PolkaETH to Bob on another chain
+
+This transfer is modelled on [Transfer via reserve](https://github.com/paritytech/xcm-format/blob/master/README.md#transfer-via-reserve).
+
+Parties:
+- H: Home chain (Polkadot-Ethereum Bridge)
+- D: Destination chain
+- R: Reserve chain
+
+Effects:
+1. H will withdraw 21 PolkaETH from Alice's local account.
+2. The sovereign account of H on R will be reduced by 21 PolkaETH.
+3. The sovereign account of D on R will be credited with 21 PolkaETH.
+4. D will mint 21 PolkaETH into Bob's account.
```

Comment:
I think Gav explained reserve-backed transfers a bit better on riot chat:

> Reserve-backed derivatives are destroyed on the chain which is sending. On the reserve chain itself, assets are transferred from the sovereign account of the sending chain to the sovereign account of the receiving chain. Reserve backed derivatives are then minted on the receiving chain, but only because the reserve chain informed the receiving chain that a deposit had been made to its sovereign account.

Note that the scenario I described is only for transfers between parachains linked by a common relay chain. So this means that:

H: Our parachain (polkadot-ethereum bridge)
R: The relay chain which holds sovereign accounts for both H & D.
D: The destination parachain.

The actual ethereum blockchain is not involved at all.

The XMCP v0 spec at https://github.com/paritytech/xcm-format describes the concept of sovereign accounts in more detail.